app/forms/CommodityCodeRulingReferenceFormProvider.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+
+class CommodityCodeRulingReferenceFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[String] =
+    Form(
+      "value" -> text("commodityCodeRulingReference.error.required")
+        .verifying(maxLength(100, "commodityCodeRulingReference.error.length"))
```

Comment:
Worth checking with Joe what length these references should be + if they should be numeric only

Reply:
These `maxLength` character limitations are auto generated by scaffold. They can be removed and adjusted where relevant.

Reply:
I think this is one the problems we will face by doing a really big Pull Request. Is not anybody fault is just we are starting from a templates and those needs some extra work to be adjust it to our requirements (business and technical).